### PR TITLE
astar_avoid: Small Cleanup

### PR DESCRIPTION
**Pull request by icolwell-as**
Transferred from GitLab. See original PR for details:
https://gitlab.com/autowarefoundation/autoware.ai/core_planning/-/merge_requests/88

### CODE REVIEW

File: waypoint_planner/src/astar_avoid/astar_avoid.cpp [hunk 1]
```diff
@@ -156,8 +125,8 @@ void AstarAvoid::run()
   // relaying mode at startup
   state_ = AstarAvoid::STATE::RELAYING;
 
-  // start publish thread
-  publish_thread_ = std::thread(&AstarAvoid::publishWaypoints, this);
```
Comment (JWhitleyWork) [inline waypoint_planner/src/astar_avoid/astar_avoid.cpp:160]: So, I understand getting rid of the thread in favor of a timer, but if the timer is just going to trigger at the same rate that the node spins, why not just call `publishWaypoints` as part of the node's main loop? The timer isn't asynchronous so you aren't gaining anything except the overhead of the timer.

**Review by JWhitleyWork — CHANGES_REQUESTED** [review]
See comment.

**Comment by JWhitleyWork** [comment]
@icolwell-as I know this is a hold-over from Gitlab and it was in limbo for quite a long time so I'm willing to let it go by this time but, in the future, please be more descriptive with your commit messages and ticket names. This "small cleanup" changes the functionality of a node in at least two different ways which can affect other's usage of this node.

File: waypoint_planner/src/astar_avoid/astar_avoid.cpp (hunk 1 above)
Comment (icolwell-as) [reply waypoint_planner/src/astar_avoid/astar_avoid.cpp:160]: Pinging @jilinzhouas since he was the original PR creator.

Comment (jilinzhouas) [reply waypoint_planner/src/astar_avoid/astar_avoid.cpp:160]: Even we are running this on the real-time operating system, the timer will publish the waypoint more consistently in terms of publishing interval.

**Pull request merged**